docs/pages/demo/datepicker/ServerRequest.example.jsx

```diff
@@ -4,23 +4,29 @@ import { TextField } from '@material-ui/core';
 import { DatePicker, Day } from '@material-ui/pickers';
 import { makeJSDateObject } from '../../../utils/helpers';
 
-function getRandomNumber(min, max) {
-  return Math.round(Math.random() * (max - min) + min);
-}
-
 function ServerRequest() {
-  const savedRequestRef = React.useRef(null);
+  const requestAbortController = React.useRef(null);
   const [selectedDays, setSelectedDays] = React.useState([1, 2, 15]);
   const [selectedDate, handleDateChange] = React.useState(new Date());
 
-  const handleMonthChange = () => {
-    window.clearTimeout(savedRequestRef.current)
-    setSelectedDays(null);
+  const handleMonthChange = date => {
+    if (requestAbortController.current) {
+      requestAbortController.current.abort();
+    }
+
+    setSelectedDays(null)
+
+    const controller = new AbortController();
+    fetch(`/fakeApi/randomDate?month=${date.toString()}`, {
+      signal: controller.signal,
+    })
+      .then(res => res.json())
+      .then(({ daysToHighlight }) => {
+        setSelectedDays(daysToHighlight);
```

Comment:
What if the component unmounts before the request resolves? I believe the set state call will warn/throw. The solution proposed in could be applied here too https://github.com/mui-org/material-ui-x/pull/5#discussion_r429676864

Reply:
We are saving abort controller. So needs to make an effect that will aborts request on unmount thanks

Reply:
It's one way to solve this. I have no idea what's the best approach. I guess it's good enough, no need to look further 👍

Reply:
Interesting, but I think that [this](https://github.com/mui-org/material-ui/blob/90babbd13d8dc624ff23453a4e47a311e2758aa6/docs/src/pages/components/autocomplete/Asynchronous.tsx#L35) solution has one giant pitfall – it doesn't abort the request. 

Probably if we want to support IE we must not use fetch and promises at all.

Reply:
Can requests be aborted? I mean, does it change something at the network layer?

Reply:
Actually request/response model cannot be aborted, but browsers can prevent some calculations on the responses if they are aborted (like processing low-level body and CORS) 

Here network log when fast switching between months with request aborts: 
![image](https://user-images.githubusercontent.com/16926049/82890265-925f2680-9f54-11ea-9f46-80fb0242d8bc.png)

Without: 
![image](https://user-images.githubusercontent.com/16926049/82890432-d5b99500-9f54-11ea-809e-567c407099bd.png)

Reply:
Nice :)